prow/plank/controller.go

```diff
@@ -530,6 +545,7 @@ func (c *Controller) syncTriggeredJob(pj prowapi.ProwJob, pm map[string]coreapi.
 			WithField("from", prevState).
 			WithField("to", pj.Status.State).Info("Transitioning states.")
 	}
+	c.managePodJobs(pj)
```

Comment:
This doesn't really make sense here, we now run a pod cleanup once for each prowjob that is triggered, but we want to run it only once.  Putting it on line 278 would probably make sense